dirsrvtests/tests/tickets/ticket2058_test.py

```diff
@@ -0,0 +1,131 @@
+    check_single_keep_alive(topo, "standalone2", f"cn=repl keep alive 1, {SUFFIX}", expected)
+    check_single_keep_alive(topo, "standalone2", f"cn=repl keep alive 2, {SUFFIX}", expected)
+
+
+def test_ticket2058(topo):
+    """Checks that replication keep alive entry are created after on line initialisations
+       (without the fix, only the entry from master1 is created)
+
+    :setup: 2 Instances
+    :steps:
+        1. Enable replication and creates agreements using dsconf
+        2  Check that keep alive entries does not exists
+        3. Initialize master2 from master1
+        4. Check that keep alive entry exists
+            (Note: without the fix only the master1 entry is created (and replicated)
+    :expectedresults:
+        1. No errors
+        2. No keep alive entry exists (replication is not working because generationIds are differents)
+        3. No errors
+        4. 2 keep alive entries exists on both instances
+    """
+
+    # Note: nether 2 masters topology nor ReplicationManager class can be used for this test 
+    # Because the setup goes too far and the issue cannot be reproduced.
+    # So the topology framework is used to create/remove/access the instances and
+    # dsconf CLI to setup the replication
+
+    master1=topo.ins["standalone1"]
+    master2=topo.ins["standalone2"]
+
+    master1.start()
+    master2.start()
+    REPLICATION_MANAGER_NAME="replmgr"
+    REPLICATION_MANAGER_DN=f"cn={REPLICATION_MANAGER_NAME},cn=config"
+    REPLICATION_MANAGER_PASSWORD="secret12"
+    TIMEOUT=30
+    
+    #Step 1: setup replication: on both instances: enable replication, creates replication manager entry and agreement toward theother instance  
+    run(f"dsconf standalone1 replication enable --suffix {SUFFIX} --role master "
+        f"--replica-id 1 --bind-dn {REPLICATION_MANAGER_DN} "
+        f"--bind-passwd {REPLICATION_MANAGER_PASSWORD}")
+    run(f"dsconf standalone2 replication enable --suffix {SUFFIX} --role master "
+        f"--replica-id 2 --bind-dn {REPLICATION_MANAGER_DN} "
+        f"--bind-passwd {REPLICATION_MANAGER_PASSWORD}")
+    run(f"dsconf standalone1 replication create-manager --name {REPLICATION_MANAGER_NAME} --passwd {REPLICATION_MANAGER_PASSWORD}")
+    run(f"dsconf standalone2 replication create-manager --name {REPLICATION_MANAGER_NAME} --passwd {REPLICATION_MANAGER_PASSWORD}")
+    run(f"dsconf standalone1 repl-agmt create --suffix {SUFFIX} "
+        f"--conn-protocol LDAP --host {master2.host} --port {master2.port} "
+        f"--bind-method SIMPLE --bind-dn {REPLICATION_MANAGER_DN} --bind-passwd {REPLICATION_MANAGER_PASSWORD}  agmt")
+    run(f"dsconf standalone2 repl-agmt create --suffix {SUFFIX} "
+        f"--conn-protocol LDAP --host {master1.host} --port {master1.port} "
+        f"--bind-method SIMPLE --bind-dn {REPLICATION_MANAGER_DN} --bind-passwd {REPLICATION_MANAGER_PASSWORD}  agmt")
```

Comment:
It's more efficient to use direct lib389 calls. It makes the code easier to debug and modify.
I see that you want to create a topology and init it later.
Check the code from this test case:
https://github.com/389ds/389-ds-base/blob/master/dirsrvtests/tests/suites/replication/regression_test.py#L195

And feel free to ping me or other team members.
lib389 documentation is a bit poor now...

Reply:
Yeah, I was going to say, don't you want a topology fixture here? There are some examples in the replication suite ....

Reply:
I think `create_topology` function is built the way so it does init during the creation implicitly.
We can write another topology fixture which doesn't do that but I think the case is very rare... So it's better to setup replication here as Pierre has done (but make it with lib389 tools as I mentioned above).

Reply:
Finally I found an easier method: using 2 master topology then reinitialize the replicas from an ldif without any replication metadata.  I already tried this method without success, because generating the ldif is not so easy (as exporting ldif without the replication data still export them) but seeing one of the testcase in regression_test have shown me how to easily filter the ldif file. So exporting the db then filter ldif to remove nsUniqueId and keepalive entries bring the replica state
 to the same state than the CLI (i.e: replication role and agreements are enabled but replication is stalled because master
  has different generationIds)